// web-kit-components/src/components/styled-components/Button.ts
import { styled } from "styled-components";

type ButtonProps = {
  borderRadius?: string;
  border?: string;
  background?: string;
  width?: string;
  height?: string;
  color?: string;
  fontFamily?: string;
  fontSize?: string;
  fontWeight?: string;
  outline?: string;
  lineHeight?: string;
  margin?: string;
  padding?: string;
  hoverColor?: string;
  activeColor?: string;
  T_ClassName?: 'small' | 'large' | 'normal' | 'medium' | string;
};
export const Button = styled.button.attrs<ButtonProps>((props) => ({
  className: props.T_ClassName,
})) <ButtonProps>`
    border-radius: ${(props) => props.borderRadius || '10px'};
    background: ${(props) => props.theme.colors.brandColors.primary || '#005792'};
    width: ${(props) => props.width || '400px'};
    height: ${(props) => props.height || '50px'};
    display: flex;
    justify-content: center;
    align-items: center;
    color: ${(props) => props.theme.colors.blackColors.white || props.color || '#fff'};
    font-family: ${(props) => props.theme.font || props.fontFamily || 'Nunito'};
    font-size: ${(props) => props.fontSize || '18px'};
    font-style: normal;
    font-weight: ${(props) => props.fontWeight || '600'};
    line-height: ${(props) => props.lineHeight || 'normal'};
    outline: ${(props) => props.outline || 'none'};
    border: ${(props) => props.border || 'none'};
    margin: ${(props) => props.margin || '3px 0'};
    padding: ${(props) => props.padding || 0};
    transition: 0.2s ease-in-out;
    white-space: nowrap;
    text-overflow: ellipsis;
    cursor: pointer;
    
    &:hover {
      background: ${(props) => props.hoverColor || 'rgba(9, 44, 76, 1)'};
    }
    &:active {
      background: ${(props) =>
    props.activeColor ||
    'linear-gradient(0deg, rgba(255, 255, 255, 0.30) 0%, rgba(255, 255, 255, 0.30) 100%), #092C4C;'};
    }
    &:disabled {
      background: ${(props) => props.hoverColor || '#E0E0E0'};
    }
  
    &.small {
      width: 241px;
      height: 48px;
      padding: ${(props) => props.padding || '14px 70px 14px 70px'};
      border-radius: ${(props) => props.borderRadius || '8px'};
  
      //styleName: Text - Small Bold;
      font-family: Inter;
      font-size: 14px;
      font-weight: 700;
      line-height: 20px;
      letter-spacing: 0em;
      text-align: left;
    }
    &.normal {
      width: 275px;
      height: 55px;
      padding: ${(props) => props.padding || '16px 80px 16px 80px'};
      border-radius: ${(props) => props.borderRadius || '8px'};
  
      //styleName: Text - Normal Bold;
      font-family: Inter;
      font-size: 16px;
      font-weight: 700;
      line-height: 22px;
      letter-spacing: 0em;
      text-align: left;
    }
    &.medium {
      width: 310px;
      height: 62px;
      padding: ${(props) => props.padding || '18px 90px 18px 90px'};
      border-radius: ${(props) => props.borderRadius || '8px'};
  
      //styleName: Text - Medium Bold;
      font-family: Inter;
      font-size: 18px;
      font-weight: 700;
      line-height: 25px;
      letter-spacing: 0em;
      text-align: left;
    }
    &.large {
      width: 344px;
      height: 68px;
      padding: ${(props) => props.padding || '20px 100px 20px 100px'};
      border-radius: ${(props) => props.borderRadius || '8px'};
  
      //styleName: Text - Large Bold;
      font-family: Inter;
      font-size: 20px;
      font-weight: 700;
      line-height: 28px;
      letter-spacing: 0em;
      text-align: left;
    }
  
    &.outline {
      border: 2px solid #092c4c;
      background-color: #fff;
      color: #000;
      transition: none;
      :hover {
        background: linear-gradient(
            0deg,
            rgba(255, 255, 255, 0.8) 0%,
            rgba(255, 255, 255, 0.8) 100%
          ),
          #092c4c;
      }
      :active {
        background: linear-gradient(
            0deg,
            rgba(255, 255, 255, 0.6) 0%,
            rgba(255, 255, 255, 0.6) 100%
          ),
          #092c4c;
      }
      :disabled {
        background-color: rgba(189, 189, 189, 1);
      }
    }

    &.LoginButton {
    font-size: 16px;
    font-weight: 400;
    transition: background 0.25s ease-in;
    border: 1px solid #005792;
    background: white;
    color: black;
    &:hover {
      background: black;
      color: white;
      box-shadow: 0px 10px 30px 0px rgba(11, 161, 235, 0.4);
      border: none;
    }
  }
  `;
