// web-kit-components/src/components/styled-components/Text.ts
import styled from "styled-components";

export type TextProps = {
  color?: string;
  fontFamily?: string;
  fontSize?: string;
  fontStyle?: string;
  fontWeight?: string;
  lineHeight?: string;
  display?: string;
  T_Level?: string;
};

export const Text = styled.span.attrs<TextProps>((props) => ({
  className: props.T_Level,
})) <TextProps>`

/* Below is for normal regular */
     color: ${(props) => props.color || props.theme.colors.blackColors.black1 || '#092c4c'};
    font-family: ${(props) => props.fontFamily || props.theme.font || 'Nunito'};
    font-size: ${(props) => props.fontSize || '14px'};
    font-style: ${(props) => props.fontStyle || 'normal'};
    font-weight: ${(props) => props.fontWeight || 400};
    line-height: ${(props) => props.lineHeight || '22.4px'};
    display: ${(props) => props.display || 'inline'};
  
    &.largeBold {
      /* Text - Large Bold */
      font-size: 18px;
      font-weight: 700;
    }
    &.largeRegular {
      /* Text - Large */
      font-size: 18px;
      font-weight: 400;
    }
    &.mediumBold {
      /* Text - Medium Bold */
      font-size: 18px;
      font-weight: 700;
    }
    &.mediumRegular {
      /* Text - Medium */
      font-size: 18px;
      font-weight: 400;
    }
    &.normalBold {
      /* Text - Normal Bold */
      font-size: 14px;
      font-weight: bold;
    }
    &.smallBold {
      /* Text - Small Bold */
      font-size: 12px;
      font-weight: bold;
    }
    &.smallRegular {
      /* Text - Small */
      font-size: 12px;
      font-weight: 400;
    }
  `;
