// web-kit-components/dist/src/components/styled-components/TextInput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextInput = void 0;
const styled_components_1 = require("styled-components");
exports.TextInput = styled_components_1.styled.input.attrs((props) => ({
    className: props.T_Class,
})) `
    color: ${(props) => props.color || '#000'};
    width: ${(props) => props.width || '400px'};
    height: ${(props) => props.height || '50px'};
    font-size: ${(props) => props.fontSize || '1.2rem'};
    font-weight: ${(props) => props.fontWeight || ''};
    border-radius: ${(props) => props.borderRadius || '10px'};
    border: ${(props) => props.border || '0.5px solid #d1d1d1'};
    border-bottom: 1px solid #d1d1d1;
    background: ${(props) => props.backgroundColor || 'transparent'};
    outline: 0;
    box-sizing: border-box;
    padding: ${(props) => props.padding || '20px'};
    font-family: ${(props) => props.theme.font || props.fontFamily || 'Nunito'};
    cursor: pointer;
  
    &::placeholder {
      color: ${(props) => props.placeHolderStyle?.color || '#c0c0c0'};
      font-family: ${(props) => props.placeHolderStyle?.fontFamily || 'Nunito'};
      font-size: ${(props) => props.placeHolderStyle?.fontSize || '16px'};
      font-style: normal;
      font-weight: ${(props) => props.placeHolderStyle?.fontWeight || 400};
      line-height: ${(props) => props.placeHolderStyle?.lineHeight || ''};
    }
    &.success{
      border: 1px solid #27AE60;
    }
    &.warning{
      border: 1px solid #E2B93B;
    }
    &.error{
      border: 1px solid #EB5757;
    }
  `;
